Updates to the readthedocs (#266)
NOAA-CSL/MELODIES-MONET: docs/appendix/troubleshooting.rst

Troubleshooting
===============

Installation problems
---------------------
    * Conda installation fails:

>         * Often the problem is in the installation of wrf-python. Check that your computer does not have an Apple Silicon CPU (Apple Intel should be fine) and that the Pyhton version is compatible with the wrf-python conda package.

Reviewer: Outermost bullet level can be all the way to left.

>     * Conda installation fails:
>         * Often the problem is in the installation of wrf-python. Check that your computer does not have an Apple Silicon CPU (Apple Intel should be fine) and that the Pyhton version is compatible with the wrf-python conda package.

Reviewer: ```suggestion
    * Conda installation fails:
        * Often the problem is in the installation of wrf-python. Check that your computer does not have an Apple Silicon CPU (Apple Intel should be fine) and that the Python version is compatible with the wrf-python Conda package.
```

 
Runtime issues
--------------

>     * analysis.pair_models() takes too long:

Reviewer: ```suggestion
    * Pairing (:meth:`~melodies_monet.driver.analysis.pair_models`) takes too long:
```
To link to the API section

Reviewer: If you won't want to link to the API section, put it in double backticks ``` ``analysis.pair_models()`` ``` to indicate that it is code.

Author: I see no problem with linking it to the API, unless you think it could be distracting

        * analysis.pair_models() is one of the most expensive functions in MELODIES-MONET, and you might be running out of memory. If you have access to more RAM, try it with that. A Time Chunking functionality is being developed to deal with this issue.

>     * The plots are not produced. The error message references LabelHandles:

Reviewer: ```suggestion
    * The plots are not produced. The error message references ``leg.legendHandles``:
```

>         * You are probably using matplotlib 3.9+ with Pandas 1.x. Downgrade matplotlib to 3.8 (upgrading Pandas should also work, but you might run into some incompatibilities for some specific functionalities, especially those related to MONETIO. Check :doc:`/getting_started/installation`).

Reviewer: ```suggestion
        * You are probably using matplotlib 3.9+ with Pandas 1.x. Downgrade matplotlib to 3.8 (upgrading Pandas should also work, but you might run into some incompatibilities for some specific functionalities, especially those related to downloading observational data with MONETIO. Check :doc:`/getting_started/installation`).
```

> 

Author: That's a good idea, I added it there.

===============

Installation problems
---------------------
* Conda installation fails:
    * Often the problem is in the installation of wrf-python. Check that your computer does not have an Apple Silicon CPU (Apple Intel should be fine) and that the Python version is compatible with the wrf-python Conda package.
 
Runtime issues
--------------
* Pairing (:meth:`~melodies_monet.driver.analysis.pair_models`) takes too long:
    * ``analysis.pair_models()`` is one of the most expensive functions in MELODIES MONET, and you might be running out of memory. If you have access to more RAM, try it with that. A Time Chunking functionality is being developed to deal with this issue. Check :doc:`/users_guide/time_chunking`.
* The plots are not produced. The error message references ``leg.legendHandles``:
    * You are probably using Matplotlib 3.9+ with pandas 1.x. Downgrade Matplotlib to 3.8 (upgrading pandas should also work, but you might run into some incompatibilities for some specific functionalities, especially those related to downloading observational data with MONETIO. Check :doc:`/getting_started/installation`).
* NaN does not exist, or pandas not detecting NaN values, etc:

>     * numpy v2.0.0 removed np.NaN, which has been completely replaced by np.nan. np.NaN is required by some versions of MONETIO. There are two solutions: (i) downgrading numpy to previous versions or (ii) upgrading MONETIO. If you go with option (ii), you might need to use the _develop_ branch of MONETIO (this should make it into _stable_ soon).

Reviewer: ```suggestion
    * In NumPy 2.x, ``np.NaN`` has been completely replaced by ``np.nan``. ``np.NaN`` is required by some versions of MONETIO. There are two solutions: (i) downgrading NumPy to previous versions (``'numpy<2'``) or (ii) upgrading MONETIO (``'monetio>=0.2.7'``). If you go with option (ii), you might need to use the _develop_ branch of MONETIO (this should make it into _stable_ soon).
```
